### PR TITLE
Initial version of systematic sensitivity analysis migration

### PR DESCRIPTION
**Pull request by murgeyseb**
Add call to systematic sensitivity computation in PowSyBl.

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
This changes migrate FARAO systematic sensitivity analysis engine to the one provided by PowSyBl, allowing better performance than current one.

**What is the current behavior?** *(You can also link to an open issue here)*
Multiple sensitivity computation and loadflow runs.

**What is the new behavior (if this is a feature change)?**
One simple call to systematic sensitivity analysis.

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*
No

### CODE REVIEW

File: util/src/main/java/com/farao_community/farao/util/FaraoNetworkPool.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            if (!isSuccess) {
+                throw new AssertionError(String.format("Cannot offer copy n°'%d' in pool. Should not happen", i + 1));
+            }
+        }
+        network.getVariantManager().setWorkingVariant(initialVariant);
+    }
+
+    public Network getAvailableNetwork() throws InterruptedException {
+        Network networkClone = networksQueue.take();
+        networkClone.getVariantManager().cloneVariant(stateSaveVariant, targetVariant, true);
```
[the remark this answers is not in the document]
Comment (murgeyseb) [reply util/src/main/java/com/farao_community/farao/util/FaraoNetworkPool.java:68]: The pool creates the variants when giving you the network clone, but the network clone variants are only manipulated before giving the network to the client, or after getting it back. Inside your client, you can do whatever you want, the network is all yours

File: util/src/main/java/com/farao_community/farao/util/FaraoNetworkPool.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        networkClone.getVariantManager().setWorkingVariant(targetVariant);
+        return networkClone;
+    }
+
+    private void cleanVariants(Network networkClone) {
+        List<String> variantsToBeRemoved = networkClone.getVariantManager().getVariantIds().stream()
+                .filter(variantId -> !variantId.equals(VariantManagerConstants.INITIAL_VARIANT_ID))
```
[the remark this answers is not in the document]
Comment (murgeyseb) [reply util/src/main/java/com/farao_community/farao/util/FaraoNetworkPool.java:75]: You cannot remove variant INITIAL_VARIANT_ID in IIDM, that's all

File: util/src/main/java/com/farao_community/farao/util/FaraoVariantsPool.java [hunk 3]
```diff
@@ -34,16 +37,21 @@ public FaraoVariantsPool(Network network, String initialVariant) {
...
-            String variantId = getVariantByIndex(i);
-            LOGGER.info("Filling variants pool with variant '{}'", variantId);
-            network.getVariantManager().cloneVariant(initialVariant, variantId);
-            boolean isSuccess = variantsQueue.offer(variantId);
-            if (!isSuccess) {
-                throw new AssertionError(String.format("Cannot offer variant '%s' in pool. Should not happen", variantId));
+        LOCK.lock();
+        try {
+            for (int i = 0; i < getParallelism(); i++) {
+                String variantId = getVariantByIndex(i);
```
[the remark this answers is not in the document]
Comment (murgeyseb) [reply util/src/main/java/com/farao_community/farao/util/FaraoVariantsPool.java:43]: perhaps something like getVariantIdByIndex would be better ?

[the remark this answers is not in the document]
Comment (TheMaskedTurtle) [reply util/src/main/java/com/farao_community/farao/util/FaraoVariantsPool.java:43]: yes !

**Pull request merged**